.github/workflows/validation.yml

```diff
@@ -21,6 +21,7 @@ jobs:
       - name: Install Python Dependencies
         uses: HassanAbouelela/actions/setup-python@setup-python_v1.4.0
         with:
+          install_args: "--with docs"
```

Comment:
Why do we need the `docs` dependencies in validation workflow? There's nothing actually checking whether the docs build so it doesn't seem like this should be necessary here.

That said, I wouldn't mind a workflow that automatically checks whether the docs build properly. To speed things up, perhaps it could run only when there are changes in the `/docs` folder, or some similar optimizations.

Reply:
> Why do we need the `docs` dependencies in validation workflow?

Pyright needs `sphinx` to correctly resolve imports in `conf.py`. [See this CI run](https://github.com/py-mine/mcstatus/actions/runs/3862801599/jobs/6584592963#step:7:20).

> I wouldn't mind a workflow that automatically checks whether the docs build properly.

Readthedocs has this feature. You just need to activate `Build docs on PRs` somewhere on the site.

Reply:
Ah, I think we can exclude `docs/` from pyright, I don't see type-hints being that important there.

Reply:
Then it needs to be excluded for flake8 too. But flake8 can found more important things, that we possibly would like to see.

Like [F401](https://www.flake8rules.com/rules/F401.html), [W601](https://www.flake8rules.com/rules/W601.html), [W602](https://www.flake8rules.com/rules/W602.html) or [W605](https://www.flake8rules.com/rules/W605.html).

Reply:
Does flake8 really mind unrecognized imports? I thought it would just treat these as 3rd party by default, as it doesn't really perform any deeper analysis like a type checker would, going through the defined accessible variables in the imported module/class/...

I don't think this will be an issue with flake8, at least not with default flake8, one of our extensions might be causing issues I suppose though. What's the flake8 error code that you see there?

Reply:
Fixed in a98b1a9b612408d3425661e6f1ba36f076e0b448.